### DISCUSSION TITLE
Spacemacs doesn't fully load on M2 macOS Sequoia with emacs@nightly cask

### DISCUSSION THREAD

**Discussion by jonhermansen** [syl20bnr/spacemacs#16833] [Q&A] [answered]
There is some error related to the `evilification` library, even though I don't use it.

Running on `develop` branch of spacemacs, spacemacs-base, "holy mode", no helm, and some other minor tweaks to my ~/.spacemacs file

I will post the exact error shortly...

**Comment by fnussbaum** [comment]
> I will post the exact error shortly...

Did this solve itself by any chance? Otherwise a backtrace could indeed be helpful. #16619 might be related.

**Comment by klochowicz** [comment]
The error is still there on the `nighty` cask (`pretest` one works).

here's my `emacs --debug-init`:  

```
Debugger entered--Lisp error: (void-variable evil-evilified-state-map)
  (define-key evil-evilified-state-map (kbd dotspacemacs-leader-key) spacemacs-default-map)
  spacemacs-bootstrap/init-evil-evilified-state()
  funcall(spacemacs-bootstrap/init-evil-evilified-state)
  (let* ((pkg-name (eieio-oref pkg ':name)) (owner (car (eieio-oref pkg ':owners)))) (spacemacs-buffer/message (format "%S -> init (%S)..." pkg-name owner)) (funcall (intern (format "%S/init-%S" owner pkg-name))))
  configuration-layer//configure-package(#<cfgl-package cfgl-package-57d35972>)
  mapc(configuration-layer//configure-package (#<cfgl-package cfgl-package-57d35941> #<cfgl-package cfgl-package-57d35972> #<cfgl-package cfgl-package-57d35988> #<cfgl-package cfgl-package-57d359f5> #<cfgl-package cfgl-package-57d359af>))
  (let (packages-to-configure) (let ((tail packages)) (while tail (let ((pkg-name (car tail))) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg ...) (spacemacs-buffer/message ...)) ((and ... ...) (spacemacs-buffer/message ...)) ((null ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) (t (let ... ...) (if ... nil ...) (cond ... ...)))) (setq tail (cdr tail))))) (setq packages-to-configure (reverse packages-to-configure)) (mapc 'configuration-layer//configure-package packages-to-configure) (mapc 'configuration-layer//post-configure-package packages-to-configure))
  configuration-layer//configure-packages-2((dotenv-mode evil-evilified-state holy-mode hybrid-mode pcre2el which-key-posframe))
  configuration-layer//configure-packages((2048-game abbrev ac-ispell academic-phrases ace-jump-helm-line ace-link ace-window add-node-modules-path adoc-mode afternoon-theme aggressive-indent alda-mode alect-themes all-the-icons ample-theme ample-zen-theme anaconda-mode anki-editor anki-mode ansi-colors anti-zenburn-theme apropospriate-theme archive-mode arduino-mode async atomic-chrome attrap auctex auctex-latexmk auto-compile auto-complete auto-dictionary auto-highlight-symbol auto-yasnippet avy badwolf-theme bat-mode better-jumper bind-key bind-map birds-of-paradise-plus-theme blacken bm bmx-mode bookmark browse-at-remote bubbleberry-theme buffer-menu bundler busybee-theme ...))
  configuration-layer//load()
  (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configuration has " "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading process!")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
  (let (changed-since-last-dump-p) (if (file-exists-p configuration-layer--last-dotspacemacs-configuration-layers-file) (progn (configuration-layer/load-file configuration-layer--last-dotspacemacs-configuration-layers-file))) (let ((layers dotspacemacs-configuration-layers)) (progn (if "Calling dotfile layers..." (progn (spacemacs-buffer/message "Calling dotfile layers..."))) (if (fboundp 'dotspacemacs/layers) (progn (condition-case err (dotspacemacs/layers) ((debug error) (configuration-layer//increment-error-count) (spacemacs-buffer/append ... t)))))) (setq dotspacemacs--configuration-layers-saved dotspacemacs-configuration-layers) (setq changed-since-last-dump-p (not (equal layers dotspacemacs-configuration-layers))) (spacemacs/dump-vars-to-file '(dotspacemacs-configuration-layers) configuration-layer--last-dotspacemacs-configuration-layers-file)) (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configuration has " "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading process!")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!")))))))
  configuration-layer/load()
  (let ((load-prefer-newer t) (please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((load-prefer-newer t) (please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
  eval-buffer(#<buffer  *load*> nil "/Users/mariusz/.emacs.d/init.el" nil t)  ; Reading at buffer position 3284
  load-with-code-conversion("/Users/mariusz/.emacs.d/init.el" "/Users/mariusz/.emacs.d/init.el" t t)
  load("/Users/mariusz/.emacs.d/init" noerror nomessage)
  #f(compiled-function () #<bytecode 0x12e013901f0c7211>)()
  #f(compiled-function () #<bytecode -0x4307d7bb01d2857>)()
  handler-bind-1(#f(compiled-function () #<bytecode -0x4307d7bb01d2857>) (error) startup--debug)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x427c44ece13a933>) #f(compiled-function () #<bytecode 0x731968ef8c3799b>) t)
  command-line()
  normal-top-level()
```

**Comment by jonhermansen** [comment]
> The error is still there on the `nighty` cask (`pretest` one works).
> 
> here's my `emacs --debug-init`:
> 
> ```
> Debugger entered--Lisp error: (void-variable evil-evilified-state-map)
>   (define-key evil-evilified-state-map (kbd dotspacemacs-leader-key) spacemacs-default-map)
>   spacemacs-bootstrap/init-evil-evilified-state()
>   funcall(spacemacs-bootstrap/init-evil-evilified-state)
>   (let* ((pkg-name (eieio-oref pkg ':name)) (owner (car (eieio-oref pkg ':owners)))) (spacemacs-buffer/message (format "%S -> init (%S)..." pkg-name owner)) (funcall (intern (format "%S/init-%S" owner pkg-name))))
>   configuration-layer//configure-package(#<cfgl-package cfgl-package-57d35972>)
>   mapc(configuration-layer//configure-package (#<cfgl-package cfgl-package-57d35941> #<cfgl-package cfgl-package-57d35972> #<cfgl-package cfgl-package-57d35988> #<cfgl-package cfgl-package-57d359f5> #<cfgl-package cfgl-package-57d359af>))
>   (let (packages-to-configure) (let ((tail packages)) (while tail (let ((pkg-name (car tail))) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg ...) (spacemacs-buffer/message ...)) ((and ... ...) (spacemacs-buffer/message ...)) ((null ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) ((not ...) (spacemacs-buffer/message ...)) (t (let ... ...) (if ... nil ...) (cond ... ...)))) (setq tail (cdr tail))))) (setq packages-to-configure (reverse packages-to-configure)) (mapc 'configuration-layer//configure-package packages-to-configure) (mapc 'configuration-layer//post-configure-package packages-to-configure))
>   configuration-layer//configure-packages-2((dotenv-mode evil-evilified-state holy-mode hybrid-mode pcre2el which-key-posframe))
>   configuration-layer//configure-packages((2048-game abbrev ac-ispell academic-phrases ace-jump-helm-line ace-link ace-window add-node-modules-path adoc-mode afternoon-theme aggressive-indent alda-mode alect-themes all-the-icons ample-theme ample-zen-theme anaconda-mode anki-editor anki-mode ansi-colors anti-zenburn-theme apropospriate-theme archive-mode arduino-mode async atomic-chrome attrap auctex auctex-latexmk auto-compile auto-complete auto-dictionary auto-highlight-symbol auto-yasnippet avy badwolf-theme bat-mode better-jumper bind-key bind-map birds-of-paradise-plus-theme blacken bm bmx-mode bookmark browse-at-remote bubbleberry-theme buffer-menu bundler busybee-theme ...))
>   configuration-layer//load()
>   (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configuration has " "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading process!")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!"))))))
>   (let (changed-since-last-dump-p) (if (file-exists-p configuration-layer--last-dotspacemacs-configuration-layers-file) (progn (configuration-layer/load-file configuration-layer--last-dotspacemacs-configuration-layers-file))) (let ((layers dotspacemacs-configuration-layers)) (progn (if "Calling dotfile layers..." (progn (spacemacs-buffer/message "Calling dotfile layers..."))) (if (fboundp 'dotspacemacs/layers) (progn (condition-case err (dotspacemacs/layers) ((debug error) (configuration-layer//increment-error-count) (spacemacs-buffer/append ... t)))))) (setq dotspacemacs--configuration-layers-saved dotspacemacs-configuration-layers) (setq changed-since-last-dump-p (not (equal layers dotspacemacs-configuration-layers))) (spacemacs/dump-vars-to-file '(dotspacemacs-configuration-layers) configuration-layer--last-dotspacemacs-configuration-layers-file)) (cond (changed-since-last-dump-p (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message "Layer list has changed since last dump.") (configuration-layer//dump-emacs)))) (spacemacs-force-dump (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "--force-dump passed on the command line or configuration has " "been reloaded, forcing a redump.")) (configuration-layer//dump-emacs)))) ((spacemacs-is-dumping-p) (configuration-layer//load)) ((and (spacemacs/emacs-with-pdumper-set-p) (spacemacs-run-from-dump-p)) (configuration-layer/message "Running from a dumped file. Skipping the loading process!")) (t (configuration-layer//load) (if (spacemacs/emacs-with-pdumper-set-p) (progn (configuration-layer/message (concat "Layer list has not changed since last time. " "Skipping dumping process!")))))))
>   configuration-layer/load()
>   (let ((load-prefer-newer t) (please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start)))))
>   (if (not (version<= spacemacs-emacs-min-version emacs-version)) (error (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (let ((load-prefer-newer t) (please-do-not-disable-file-name-handler-alist nil)) (require 'core-spacemacs) (spacemacs/dump-restore-load-path) (configuration-layer/load-lock-file) (spacemacs/init) (configuration-layer/stable-elpa-init) (configuration-layer/load) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (spacemacs/dump-eval-delayed-functions) (if (and dotspacemacs-enable-server (not (spacemacs-is-dumping-p))) (progn (require 'server) (if dotspacemacs-server-socket-dir (progn (setq server-socket-dir dotspacemacs-server-socket-dir))) (if (server-running-p) nil (message "Starting a server...") (server-start))))))
>   eval-buffer(#<buffer  *load*> nil "/Users/mariusz/.emacs.d/init.el" nil t)  ; Reading at buffer position 3284
>   load-with-code-conversion("/Users/mariusz/.emacs.d/init.el" "/Users/mariusz/.emacs.d/init.el" t t)
>   load("/Users/mariusz/.emacs.d/init" noerror nomessage)
>   #f(compiled-function () #<bytecode 0x12e013901f0c7211>)()
>   #f(compiled-function () #<bytecode -0x4307d7bb01d2857>)()
>   handler-bind-1(#f(compiled-function () #<bytecode -0x4307d7bb01d2857>) (error) startup--debug)
>   startup--load-user-init-file(#f(compiled-function () #<bytecode 0x427c44ece13a933>) #f(compiled-function () #<bytecode 0x731968ef8c3799b>) t)
>   command-line()
>   normal-top-level()
> ```

Thanks for posting it.

**Comment by fnussbaum** [comment]
Thanks! I think this might be an issue with the build, I don't see how Spacemacs would cause this. But it is hard to tell without being able to reproduce it.

Note that in https://github.com/syl20bnr/spacemacs/issues/16619, the error went away after rebuilding with `make bootstrap`. I've also just built Emacs 31.0.50 from this commit https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=29c6dad78db6d599108612f120cd0fb1fb18e105 and could not reproduce the above error.

**Comment by klochowicz** [comment, marked as answer]
indeed, it's not Spacemac's fault. until `emacs@nightly` is fixed upstream, I can recommend using these (working) versions:

- `emacs@pretest` - if older emacs is acceptable

- emacs-plus (if you want the latest)

```sh
brew tap d12frosted/emacs-plus
 brew install emacs-plus@31 --with-native-comp
```
